src/ape/contracts/base.py

```diff
@@ -559,7 +611,47 @@ def poll_logs(
             yield from self.range(new_block.number, stop=new_block.number + 1)
 
 
-class ContractInstance(BaseAddress):
+class ContractTypeWrapper(ManagerAccessMixin):
+    contract_type: ContractType
+
+    def decode_input(self, calldata: bytes) -> Tuple[str, Dict[str, Any]]:
+        """
+        Decode the given calldata using this contract.
+        If the calldata has a method ID prefix, Ape with detect it and find
+        the corresponding method. Else, it not be able to succeed.
```

Comment:
```suggestion
    def decode_input(self, calldata: bytes) -> Tuple[str, Dict[str, Any]]:
        """
        Decode the given calldata using this contract.
        If the calldata has a method ID prefix, Ape will detect it and find
        the corresponding method, else it will error.
```